staker/rollup_watcher.go

```diff
@@ -165,6 +165,44 @@ func (r *RollupWatcher) LookupNode(ctx context.Context, number uint64) (*NodeInf
 	}, nil
 }
 
+func (r *RollupWatcher) LookupNodeByBlockNumber(ctx context.Context, blockNumber uint64) (*NodeInfo, error) {
```

Comment:
this code does something else.. which is useful for later stage.
It gives you information from an assertion in a specific block.
Later, we'll have a remote server say something "assertion node N is finalized with blockhash H in parent-chain block P". We'll need to check this assertion, and perhaps get more data - and this function is great for that..

Reply:
Ah, thanks for catching this, removed for now in that case.